src/org/labkey/test/components/ui/grids/EditableGrid.java

```diff
@@ -595,9 +611,51 @@ public EditableGrid pasteFromCell(int row, String columnName, String pasteText)
         // ... or for a second and a half
         WebDriverWrapper.waitFor(()-> (getRowCount() > initialRowCount || !indexValue.equals(gridCell.getText())) &&
                         isInSelection(gridCell), 1500);
+        if (validate)
+            waitForAnyPasteContent(pasteText);
+
         return this;
     }
 
+    /**
+     * Awaits any elements (except for empty, or space-only) of the pasted content to be present in the grid
+     * @param pasteContent  tab-separated text
+     */
+    protected void waitForAnyPasteContent(String pasteContent)
+    {
+        // split pasteContent into its parts
+        var contentParts = pasteContent.replace("\n", "\t").split("\t");
+        // filter out empty and space-only values
+        var filteredParts = Arrays.stream(contentParts).filter(a-> !a.isEmpty() && !a.equals(" ")).collect(Collectors.toList());
```

Comment:
This list is not going to be modified so you can just call .toList here. 
`var filteredParts = Arrays.stream(contentParts).filter(a-> !a.isEmpty() && !a.equals(" ")).toList());`